mypy/reachability.py

```diff
@@ -92,7 +92,7 @@ def infer_condition_value(expr: Expression, options: Options) -> int:
     else:
         result = consider_sys_version_info(expr, pyversion)
         if result == TRUTH_VALUE_UNKNOWN:
-            result = consider_sys_platform(expr, options.platform)
+            result = consider_sys_platform(expr, options.sys_platform)
```

Comment:
You seem not to be passing `options.platform_system` here, but you added it as an argument below.